### PR TITLE
[BEAM-1414] Replaces CountingInput with style guide compliant GenerateSequence

### PR DESCRIPTION
**Pull request by jkff**
Migration guide:
`CountingInput.unbounded()` -> `GenerateSequence.from(0)`
`CountingInput.upTo(n)` -> `GenerateSequence.fromTo(0, n)`

Removing rogue direct uses of CountingSource is for another PR and less critical.

### CODE REVIEW

**Comment by jkff** [comment]
R: @dhalperi

**Comment by jkff** [comment]
Dan is swamped with stuff. R: @tgroh instead.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/GenerateSequence.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    checkArgument(from >= 0, "Value of from must be non-negative, but was: %s", from);
+    return new AutoValue_GenerateSequence.Builder()
+        .setFrom(from)
+        .setTo(-1)
+        .setElementsPerPeriod(0)
+        .build();
+  }
+
+  /** Specifies to generate the range [from, to). */
+  public static GenerateSequence fromTo(long from, long to) {
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/GenerateSequence.java]: `between`?

Then you get the more readable `Generate Sequence from x to y` and `Generate Sequence between x and y`

Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/GenerateSequence.java]: Replaced with .from().to().

**Pull request closed without merging**